On Windows, fix graphics bugs after external monitor connect/disconnect

eirikbakke opened the pull request:
On Windows 10 and 11, I have observed various graphics bugs that appear after changes in the OS graphics configuration, such as connecting or disconnecting external monitors, or changing the HiDPI scaling levels. In some cases, the NetBeans main window ceased to update its graphics at all, making the IDE completely unusable until restarted, or until "Full Screen" was toggled via keyboard shortcut (a workaround I discovered after months of experimentation). In other cases, the NetBeans window might appear off-screen or partially off-screen, or the NetBeans window might have paint errors until it is maximized/restored. These are probably OpenJDK bugs or even bugs in Windows graphics drivers, as I saw similar problems in jEdit (another Java Swing app) and Skype (not a Java app).

I ended up developing the patch in this PR, which detects changes in the graphics environment on Windows and refreshes the main window in a way that resets the window state in these cases. This patch was developed and verified to work through many months of experimentation, and I have now been running it on my working NetBeans IDE for 18 months.

neilcsmith-net commented:
This makes sense but keeps bumping through milestones.  @DevCharly @mbien any thoughts?

eirikbakke commented:
> The problem I see is: When could this workaround be removed?

The answer is as you expect; never! :-) The bug is not reproducible, nor can I tell where it came from or whether it was fixed. But it did make NetBeans and my platform app completely unresponsive, on many occasions, and the fix made it go away.

As I have now switched to MacOS for my daily work, and have not been using NetBeans _without_ the patch for years, I am also unable to tell if the problem still exists.

In my commercial app, I'll be leaving the patch in forever. But no need to push it into NetBeans if it smells too bad... just wanted to offer it up before I switched to MacOS.
